static/js/hashchange.js

```diff
@@ -269,7 +270,29 @@ function hashchanged_overlay(old_hash) {
     state.old_overlay_group = get_hash_group(base);
 }
 
+exports.update_browser_history = function (new_hash) {
+    var old_hash = window.location.hash;
+
+    if (old_hash === new_hash) {
+        // If somebody is calling us with the same hash we already have, it's
+        // probably harmless, and we just ignore it.  But it could be a symptom
+        // of disorganized code that's prone to an infinite loop of repeatedly
+        // assigning the same hash.
+        blueslip.info('ignoring probably-harmless call to update_browser_history: ' + new_hash);
+        return;
+    }
+
+    state.old_hash = old_hash;
+    window.location.hash = new_hash;
+    state.is_internal_change = true;
```

Comment:
What's the story with the order-of-operations here?  I'd have expected one would need to set `is_internal_change` before doing the `window.location.hash` edit line...